drivers/pwm/pwm_renesas_ra8.c

```diff
@@ -0,0 +1,584 @@
+					       GPT_SOURCE_NONE);
+
+			data->extend_cfg.capture_a_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+		} else {
+			data->extend_cfg.start_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_RISING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+
+			data->extend_cfg.capture_a_source =
+				(gpt_source_t)(GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_LOW |
+					       GPT_SOURCE_GTIOCA_FALLING_WHILE_GTIOCB_HIGH |
+					       GPT_SOURCE_NONE);
+		}
+	}
+
+	data->capture.callback = cb;
+	data->capture.user_data = user_data;
+	data->capture.continuous = (flags & PWM_CAPTURE_MODE_CONTINUOUS) ? true : false;
+
+	if (data->capture.continuous == true) {
+		data->extend_cfg.stop_source = data->extend_cfg.capture_a_source;
+		data->extend_cfg.clear_source = data->extend_cfg.start_source;
+	} else {
+		data->extend_cfg.stop_source = (gpt_source_t)(GPT_SOURCE_NONE);
+		data->extend_cfg.clear_source = (gpt_source_t)(GPT_SOURCE_NONE);
+	}
+
+	return 0;
+}
+
+static int pwm_ra8_enable_capture(const struct device *dev, uint32_t pin)
+{
+	struct pwm_ra8_data *data = dev->data;
+	fsp_err_t err = FSP_SUCCESS;
+
+	if (pin != GPT_IO_PIN_GTIOCA) {
+		LOG_ERR("Feature only support for gtioca");
+		return -EINVAL;
+	}
+
+	if (data->capture.is_busy) {
+		LOG_ERR("Capture already active on this pin");
+		return -EBUSY;
+	}
+
+	if (!data->capture.callback) {
+		LOG_ERR("PWM capture not configured");
+		return -EINVAL;
+	}
+
+	data->capture.is_busy = true;
+
+	/* Enable capture source */
+	err = R_GPT_Enable(&data->fsp_ctrl);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+
+	/* Enable interruption */
+	enable_irq(data->fsp_cfg.cycle_end_irq, data->fsp_cfg.cycle_end_irq, &data->fsp_ctrl);
+	enable_irq(data->extend_cfg.capture_a_irq, data->extend_cfg.capture_a_ipl, &data->fsp_ctrl);
+
+	R_ICU->IELSR[data->fsp_cfg.cycle_end_irq] = (elc_event_t)data->overflow_event;
+	R_ICU->IELSR[data->extend_cfg.capture_a_irq] = (elc_event_t)data->capture_a_event;
+
+	return 0;
+}
+
+static int pwm_ra8_disable_capture(const struct device *dev, uint32_t pin)
+{
+	struct pwm_ra8_data *data = dev->data;
+	fsp_err_t err = FSP_SUCCESS;
+
+	if (pin != GPT_IO_PIN_GTIOCA) {
+		LOG_ERR("Feature only support for gtioca");
+		return -EINVAL;
+	}
+	data->capture.is_busy = false;
+
+	/* Disable interruption */
+	disable_irq(data->fsp_cfg.cycle_end_irq);
+	disable_irq(data->extend_cfg.capture_a_irq);
+
+	R_ICU->IELSR[data->fsp_cfg.cycle_end_irq] = (elc_event_t)ELC_EVENT_NONE;
+	R_ICU->IELSR[data->extend_cfg.capture_a_irq] = (elc_event_t)ELC_EVENT_NONE;
+
+	/* Disable capture source */
+	err = R_GPT_Disable(&data->fsp_ctrl);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+
+	/* Stop timer */
+	err = R_GPT_Stop(&data->fsp_ctrl);
+
+	/* Clear timer */
+	err = R_GPT_Reset(&data->fsp_ctrl);
```

Comment:
no error check?

Reply:
I added the error check here and for the others as well. Thank you very much